### ISSUE TITLE
Lockwise remains stuck when trying to re-login after a disconnect(from settings)

### ISSUE THREAD

**Issue by abodea** [mozilla-lockwise/lockwise-android#1013]
## Prerequisites 
Follow the steps from #1012 
## Steps to reproduce
1. Tap on the Get started in order to re-login.
2. Add valid credentials.
3. Tap Sign in.
### Expected behavior
The login is successfully made and the entries section can be reached without problems. 
### Actual behavior
After tapping on the Sign in button, nothing happens and Lockwise is stuck on that screen.
### Device & build information
* Device: ?
* Build version: ?
### Notes
**Please note that I cannot reproduce this issue on the previous RC(3.0.0 (Build 5132))**
Note that this is not working not even after a restart, the app is forever stuck...until deleting the data and trying to re-login.
Attachments: 
[Video](https://drive.google.com/file/d/1GSpqO1wFwhY_uwGEDbYvTsUVH3gF6r40/view?usp=sharing)
[launchLog.txt](https://github.com/mozilla-lockwise/lockwise-android/files/3754381/launchLog.txt) - Launch Logcat

**Comment by abodea** [comment]
```Kotlin
2019-10-22 11:24:32.842 25344-25344/? W/cr_AwAutofillManager: WebView autofill is disabled because WebView isn't created with activity context.
2019-10-22 11:24:33.210 1606-1606/? I/PhenotypeExperimentConfiguration: refreshConfiguration() : Force = false : UpdateAvailable = false : Age = 94 minutes : MaxAge = 720 minutes
2019-10-22 11:24:33.211 1606-1606/? I/AndroidIME: PasswordIme.onDeactivate()
2019-10-22 11:24:33.211 25344-25454/? W/IInputConnectionWrapper: requestCursorAnchorInfo on inactive InputConnection
2019-10-22 11:24:37.118 25344-25369/? I/FirefoxAccount: Executing: complete oauth flow
2019-10-22 11:24:37.257 25344-25344/? E/mozilla.lockbo: No package ID ff found for ID 0xffffffff.
2019-10-22 11:24:39.154 25344-25369/? W/WrappingPersistenceCallback: InternalFxAcct tried persist state, but persistence callback is not set
2019-10-22 11:24:39.155 25344-25369/? I/FirefoxAccount: Successfully executed: complete oauth flow
2019-10-22 11:24:39.205 870-870/? I/keystore: 1 0
2019-10-22 11:24:39.235 1162-1177/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
2019-10-22 11:24:39.235 1162-1177/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
2019-10-22 11:24:39.238 25344-25468/? I/FirefoxAccount: Executing: getProfile
2019-10-22 11:24:39.251 25344-25369/? I/FirefoxAccount: Executing: get access token
2019-10-22 11:24:40.277 790-972/? D/CHRE: @ 89482.133: [ImuCal] Dynamic sensor configuration: stand-by.
2019-10-22 11:24:40.398 25344-25468/? W/WrappingPersistenceCallback: InternalFxAcct tried persist state, but persistence callback is not set
2019-10-22 11:24:40.399 25344-25468/? I/FirefoxAccount: Successfully executed: getProfile
2019-10-22 11:24:40.646 25344-25369/? I/FirefoxAccount: Successfully executed: get access token
2019-10-22 11:24:41.713 870-870/? I/keystore: 1 0
2019-10-22 11:24:41.740 870-870/? I/keystore: 1 1
2019-10-22 11:24:41.763 1162-1177/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
2019-10-22 11:24:41.763 1162-1177/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
2019-10-22 11:24:41.773 1162-1177/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
2019-10-22 11:24:41.773 1162-1177/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
2019-10-22 11:24:41.776 25344-25471/? D/RustNativeSupport: findMegazordLibraryName(logins, 0.37.1
2019-10-22 11:24:41.776 25344-25471/? D/RustNativeSupport: lib in use: none
2019-10-22 11:24:41.776 25344-25471/? D/RustNativeSupport: lib configured: megazord
2019-10-22 11:24:41.776 25344-25471/? D/RustNativeSupport: lib version configured: 0.37.1
2019-10-22 11:24:41.776 25344-25471/? D/RustNativeSupport: settled on megazord
2019-10-22 11:24:43.794 25344-25344/? E/mozilla.lockbo: No package ID ff found for ID 0xffffffff.
2019-10-22 11:24:43.795 25344-25344/? I/chatty: uid=10516(mozilla.lockbox) identical 1 line
2019-10-22 11:24:43.795 25344-25344/? E/mozilla.lockbo: No package ID ff found for ID 0xffffffff.
2019-10-22 11:24:44.748 790-972/? D/CHRE: @ 89486.609: [ImuCal] Dynamic sensor configuration: stand-by.
2019-10-22 11:24:44.831 790-972/? D/CHRE: @ 89486.688: [ImuCal] Dynamic sensor configuration: high-performance.
2019-10-22 11:24:44.831 790-972/? D/CHRE: @ 89486.695: [ImuCal] Dynamic sensor configuration: stand-by.
2019-10-22 11:24:45.223 790-1233/? I/sensors-hal: update_qtimer_to_realtime_offset:192, updating qtimer-realtime offset (gap 6042), offset_diff = 16934
2019-10-22 11:24:49.053 790-972/? I/CHRE: @ 89490.914: [ImuCal] [NanoSensorCal:GYRO_RPS] Offset | Temperature [C]: -0.000842, 0.000596, 0.000180 | 31.38
2019-10-22 11:24:49.054 790-972/? I/CHRE: @ 89490.914: [ImuCal] [NanoSensorCal:GYRO_RPS] Temp Sensitivity: 0.000081, -0.000006, -0.000029
2019-10-22 11:24:49.054 790-972/? I/CHRE: @ 89490.914: [ImuCal] [NanoSensorCal:GYRO_RPS] Temp Intercept: -0.003397, 0.000800, 0.001111
2019-10-22 11:24:50.376 790-972/? D/CHRE: @ 89492.234: [ImuCal] Dynamic sensor configuration: stand-by.
2019-10-22 11:24:55.959 790-972/? D/VSC: @ 89497.812: [Significant Motion] Request motion_detect, interval 18446744073709551615 ns, latency 0 ns
2019-10-22 11:24:55.959 790-972/? D/VSC: @ 89497.812: [Significant Motion] Stop accel
2019-10-22 11:24:55.959 790-972/? I/CHRE: @ 89497.820: [AR_CHRE] still: 56
2019-10-22 11:24:57.495 1759-1759/? D/QtiCarrierConfigHelper: WARNING, no carrier configs on phone Id: 0
2019-10-22 11:24:59.164 790-972/? I/CHRE: @ 89501.023: [AR_CHRE] still: 100

**Comment by abodea** [comment]
```Kotlin
2019-10-22 11:27:58.121 25344-25369/? I/FirefoxAccount: Executing: complete oauth flow
2019-10-22 11:27:58.123 25344-25369/? E/FirefoxAccount: Error while running: complete oauth flow
    c.c.b.d$d: Unknown OAuth State
        at c.c.b.o.b.intoException(RustError.kt:4)
        at c.c.b.c.a(FirefoxAccount.kt:21)
        at c.b.a.a.d$a.invoke(FirefoxAccount.kt:3)
        at c.b.a.a.k.invoke(Utils.kt:1)
        at h.a.a.a.v0.l.v0.a(TypeWithEnhancement.kt:75)
        at h.a.a.a.v0.l.v0.a(TypeWithEnhancement.kt:210)
        at c.b.a.a.d.c(FirefoxAccount.kt:5)
        at h.y.i.a.a.b(ContinuationImpl.kt:2)
        at i.a.m0.run(Dispatched.kt:15)
        at i.a.b2.b.a(CoroutineScheduler.kt:74)
        at i.a.b2.b$a.run(CoroutineScheduler.kt:45)
2019-10-22 11:27:58.156 870-870/? I/keystore: 1 0
2019-10-22 11:27:58.180 1162-1177/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
2019-10-22 11:27:58.180 1162-1177/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
2019-10-22 11:27:58.214 1162-1177/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
2019-10-22 11:27:58.214 1162-1177/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
2019-10-22 11:27:58.206 870-870/? I/chatty: uid=1017(keystore) /system/bin/keystore identical 1 line
2019-10-22 11:27:58.234 870-870/? I/keystore: 1 0
2019-10-22 11:27:58.239 25344-25469/? I/FirefoxAccount: Executing: getProfile
2019-10-22 11:27:58.240 25344-25344/? E/mozilla.lockbo: No package ID ff found for ID 0xffffffff.
2019-10-22 11:27:58.243 1162-1177/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
2019-10-22 11:27:58.243 1162-1177/? W/BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
2019-10-22 11:28:00.263 25344-25469/? W/WrappingPersistenceCallback: InternalFxAcct tried persist state, but persistence callback is not set
2019-10-22 11:28:00.264 25344-25469/? I/FirefoxAccount: Successfully executed: getProfile
2019-10-22 11:28:02.061 790-972/? D/CHRE: @ 89683.922: [ImuCal] Dynamic sensor configuration: stand-by.
2019-10-22 11:28:02.087 790-972/? W/ASH: @ 89683.945: bmi160: timestamp decrement on sensor [1]: prev = 89540,418,200,012, curr = 89540,408,410,128
2019-10-22 11:28:02.571 25344-25344/? E/mozilla.lockbo: No package ID ff found for ID 0xffffffff.
2019-10-22 11:28:02.571 25344-25344/? I/chatty: uid=10516(mozilla.lockbox) identical 1 line
2019-10-22 11:28:02.572 25344-25344/? E/mozilla.lockbo: No package ID ff found for ID 0xffffffff.
2019-10-22 11:28:06.641 790-972/? I/CHRE: @ 89688.492: [AR_CHRE] still: 34
2019-10-22 11:28:08.370 790-972/? D/CHRE: @ 89690.234: [ImuCal] Dynamic sensor configuration: stand-by.
```

**Comment by abodea** [comment]
Verified as fixed on **`v3.0.1 (Build 5124)`** using Google Pixel 3a XL(A 9), Samsung Galaxy S10+(A 9).
Please note that I did at least 15 disconnects + re-login and everything worked as expected.
Sometimes when tapping on the disconnect button, it take few seconds(~2-5) until the Welcome Screen with the `Get Started` button is displayed.
@eliserichards @jhugman looks good 🥳!
